sql/catalyst/src/main/java/org/apache/spark/sql/connector/expressions/Cast.java

```diff
@@ -29,14 +29,31 @@
 @Evolving
 public class Cast extends ExpressionWithToString {
   private Expression expression;
+
+  /**
+   * Original data type of given expression
+   */
+  private DataType expressionDataType;
+
+  /**
+   * Target data type, i.e. data type in which expression will be cast
+   */
   private DataType dataType;
 
+  @Deprecated
```

Comment:
@cloud-fan Since Cast is still evolving, can we delete this constructor or it is better to leave it as it is

Reply:
The cost to keep it is low, it's fine to leave it

Reply:
let's explicitly set `expressionDataType` to null in this constructor.

Reply:
I agree with you